src/peep/peep.c

```diff
@@ -610,19 +611,24 @@ static uint8 peep_assess_surroundings(sint16 center_x, sint16 center_y, sint16 c
 					break;
 				case MAP_ELEMENT_TYPE_TRACK:
 					ride = get_ride(mapElement->properties.track.ride_index);
+					if (ride->lifecycle_flags & RIDE_LIFECYCLE_MUSIC &&
+						ride->status != RIDE_STATUS_CLOSED &&
+						!(ride->lifecycle_flags & (RIDE_LIFECYCLE_BROKEN_DOWN | RIDE_LIFECYCLE_CRASHED))){
+						music_on = true;
+					}
```

Comment:
All the if blocks below this line can be moved into the if block above removing the need for `music_on`